### DISCUSSION TITLE
is it possible to translate NavigationGroups

### DISCUSSION THREAD

**Discussion by G3z** [filamentphp/filament#2295] [Help] [answered]
I currently register navigation groups this way, is it possible to translate them ?
```php
Filament::registerNavigationGroups([
  'Sales',
  'Diamonds',
  'Colored Stones',
  'Jewels',
  'Frames',
  'Products',
  'Orders',
  'Backoffice',
  'Settings',
]);
```

if not, maybe it should be possible to use a keyed array like `['sales' => __('navigation.labels.sales')]`

**Comment by danharrin** [comment, marked as answer]
Use the advanced nav API instead of `$navigationGroup` - https://filamentphp.com/docs/2.x/admin/navigation#advanced-navigation-customization

**Comment by dev3k** [comment]
there is another way

```php
Filament::registerNavigationGroups([
     __('Settings')
]);

//in resource add 
protected static function getNavigationGroup(): ?string
{
    return __('Settings');
}
```

**Reply by markosole** [reply]
Just small correction, protected is failing and working example is:

```php

public static function getNavigationGroup(): ?string
    {
        return __('Settings');
    }

```
I am looking how to do the same for other things like modelLabel etc. Getting there, Filament is great!

**Reply by dev3k** [reply]
I think "protected" from v2, but not sure, for modelLabel 

```php
public static function getModelLabel(): string
{
    return __('Item');
}
    
public static function getPluralModelLabel(): string
{
    return __('Items');
}
```
in your resource file

**Reply by danharrin** [reply]
All of this can be found in the docs btw - https://filamentphp.com/docs/3.x/panels/resources/getting-started#customizing-the-model-label

**Reply by markosole** [reply]
Thank you Dan, I've figured out that there was not just the `public` and `protected` but also `getModelLabel(): ? string`   and  latest `getModelLabel(): string` with no `?` before string. 

There is a lot on docs, a lot I mean a lot on small place :) And there are so many different possibilities which is great and mad at the same time :) I love it! Well designed and architected.

**Comment by LukasCCB** [comment]
Why `__('System")` not works? Is reserved ?

`App\Filament\Admin\Resources\ChangelogResource::getNavigationGroup(): Return value must be of type ?string, array returned`

But, any other value like __('System") or __('Bla Bla") works.

**Comment by markosole** [comment]
@LukasCCB  I've ran into same issue weeks ago, and found a way to do it. Here is an example of `AdminPanelProvider`.
Notice the `->label` property where funtion returns a string of resolved "translated" value. 

```php

return $panel
->default()
->id('admin')
->path('admin')
->login()
->navigationItems([
    NavigationItem::make('Analytics')
        ->url('/analitika', shouldOpenInNewTab: true)
        ->icon('heroicon-o-presentation-chart-line')
        ->group('Analitika')
        ->sort(3),
    NavigationItem::make('drafts')
        ->label(fn (): string => __('Manager Users')) // YOUR SINGLE MENU ITEM NAME
        ->url('/admin/admin-users/', shouldOpenInNewTab: false)
        ->icon('heroicon-o-briefcase')
        ->group(fn (): string => __('User Management')) // YOUR GROUP NAME
        ->badge(fn (): string => self::numberManagerUsers())                    
        ->sort(3),
    NavigationItem::make('dashboard')
        ->label(fn (): string => __('filament-panels::pages/dashboard.title'))
        ->url(fn (): string => Dashboard::getUrl())
        ->icon('heroicon-o-presentation-chart-line')
        ->isActiveWhen(fn () => request()->routeIs('filament.admin.pages.dashboard')),
])
            
 ```
 
 Same works for `self::numberManagerUsers()` which is a function defined at the end of file:
 
 ```php
 public static function numberManagerUsers(): string
    {
        return User::where('is_manager', 1)->count();
    }
    
```

I do not know why is this the case but works well for me on version 3.1x
